File: test/fixtures/js-capture/shipping.js
// A function whose argument is a nested object — the synthetic fuzz battery
// can't construct it, so characterization finds nothing. The example driver
// captures real boundary I/O and makes it verifiable.

function totalPrice(order) {
  let subtotal = 0;
  for (const item of order.items) subtotal += item.price * item.qty;
  return Math.round(subtotal * (1 - order.discount) * 100) / 100;
}

class Cart {
  lineCount(order) {
    return order.items.length;
  }
}

module.exports = { totalPrice, Cart };

File: test/fixtures/py-capture/shipping.py
"""A function whose argument is a nested dict — the synthetic fuzz battery
(primitives/lists) can't construct it, so characterization finds nothing.
Running the example driver captures real boundary I/O and makes it verifiable."""


def total_price(order):
    subtotal = 0
    for item in order["items"]:
        subtotal += item["price"] * item["qty"]
    return round(subtotal * (1 - order["discount"]), 2)


class Cart:
    def line_count(self, order):
        return len(order["items"])
